pkg/kubernetes/kubernetes.go

```diff
@@ -86,6 +86,7 @@ func GetKubeConfig(ctx context.Context) (*restclient.Config, error) {
 			log.Errorf("InClusterConfig failed %v", err)
 			return nil, err
 		}
+		config.QPS, config.Burst = getClientThroughput(ctx, true)
```

Comment:
What's the reason for this not being outside the `if-else` path? We would want it irrespective of config right?

Reply:
+1

Reply:
`getClientThroughput` has argument `inClusterClient` as true or false. since we are using in-cluster config here, so placed in that block.

with `false` method expect use to use diff parameters

```
	// EnvSupervisorClientQPS  is the QPS for all clients to the supervisor cluster API server
	EnvSupervisorClientQPS = "SUPERVISOR_CLIENT_QPS"

	// EnvSupervisorClientBurst  is the Burst for all clients to the supervisor cluster API server
	EnvSupervisorClientBurst = "SUPERVISOR_CLIENT_BURST"
```

Let me see if this is used by SV cluster.

Reply:
I am refatoring `func getClientThroughput(ctx context.Context, inClusterClient bool) (float32, int) {` to make it indepdenct of supervisor specific env variables, so that i can utlize this in vanilla cluster.

Reply:
here is the refactored code - https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/389/commits/dc3c0bb085af4685ec24337f82d4f22b3c6937ec

I have also considered a case when CSI pod needs to run as the process, we do not need to specify two diff env variables for in cluster and out of the cluster. 

`VSPHERE_CSI_K8S_CLIENT_QPS` is sufficient for both cases.

Reply:
Refactored code as per offline discussion - https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/389/commits/29cfa7b22adcdb3432c7fabc97967304bd9a844e